src/app/child-dev-project/attendance/attendance.service.ts

```diff
@@ -193,10 +204,54 @@ export class AttendanceService {
   }
 
   async getActivitiesForChild(childId: string): Promise<RecurringActivity[]> {
-    return await this.dbIndexing.queryIndexDocs(
+    const promiseByParticipant = this.dbIndexing.queryIndexDocs(
       RecurringActivity,
       "activities_index/by_participant",
       childId
     );
+    const visitedSchools = await this.childrenService.queryRelationsOf(
+      "child",
+      childId
+    );
+    const promisesBySchool = visitedSchools
+      .filter((schoolRelation) => schoolRelation.isActive())
+      .map((relation) =>
+        this.dbIndexing.queryIndexDocs(
+          RecurringActivity,
+          "activities_index/by_school",
+          relation.schoolId
+        )
+      );
+    const uniqueActivities = await promiseByParticipant;
+    const activitiesBySchool = await Promise.all(promisesBySchool);
```

Comment:
I am not sure whether it's worth to defer the resolving of promises for a Promise.all - it makes the code significantly harder to read. Even though these could run in parallel they are probably not really long running tasks

Reply:
Yes this function is horrible. But there are not many options to resolve an array of promises and I don't think the `Promise.all` call is the problem here. Rather the concatenation of the arrays that makes it difficult to understand.

Reply:
true, but you could skip `const uniqueActivities = await promiseByParticipant;` and use the await on the first line already